src/sql/workbench/api/common/extHostModelViewDialog.ts

```diff
@@ -640,31 +660,41 @@ class ModelViewDashboardImpl implements azdata.window.ModelViewDashboard {
 		return this._editor.closeEditor();
 	}
 
-	createTab(tab: azdata.DashboardTab, view: azdata.ModelView): azdata.Tab {
-		if (tab.toolbar) {
+	createTab(dashboardTab: DisposableDashboardTab, view: azdata.ModelView): DisposableTab {
+		let tab: DisposableTab;
+		if (dashboardTab.toolbar) {
 			const flexContainer = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
-			flexContainer.addItem(tab.toolbar, { flex: '0 0 auto' });
-			flexContainer.addItem(tab.content, { flex: '1 1 auto' });
-			return {
-				title: tab.title,
-				id: tab.id,
+			flexContainer.addItem(dashboardTab.toolbar, { flex: '0 0 auto' });
+			flexContainer.addItem(dashboardTab.content, { flex: '1 1 auto' });
+			tab = {
+				title: dashboardTab.title,
+				id: dashboardTab.id,
 				content: flexContainer,
-				icon: tab.icon
+				icon: dashboardTab.icon,
+				dispose: () => {
+					dashboardTab.toolbar.dispose();
+					dashboardTab.content.dispose();
+				},
 			};
 		} else {
-			return tab;
+			tab = dashboardTab;
 		}
+		this._register(tab);
+		return tab;
 	}
 
-	createTabs(dashboardTabs: (azdata.DashboardTab | azdata.DashboardTabGroup)[], view: azdata.ModelView): (azdata.TabGroup | azdata.Tab)[] {
-		const tabs: (azdata.TabGroup | azdata.Tab)[] = [];
-		dashboardTabs.forEach((item: azdata.DashboardTab | azdata.DashboardTabGroup) => {
+	createTabs(dashboardTabs: (DisposableDashboardTab | DisposableDashboardTabGroup)[], view: azdata.ModelView): (DisposableTabGroup | DisposableTab)[] {
+		const tabs: (DisposableTabGroup | DisposableTab)[] = [];
+		dashboardTabs.forEach((item: DisposableDashboardTab | DisposableDashboardTabGroup) => {
 			if ('tabs' in item) {
-				tabs.push(<azdata.TabGroup>{
+				tabs.push(<DisposableTabGroup>{
 					title: item.title,
 					tabs: item.tabs.map(tab => {
 						return this.createTab(tab, view);
-					})
+					}),
+					dispose: () => {
+						item.tabs.forEach(t => t.dispose());
```

Comment:
This should be disposing the tab objects created from this.createTab - not the ones passed in (since those aren't guaranteed to have dispose implemented)